watertap/tools/parameter_sweep.py

```diff
@@ -376,19 +378,23 @@ def _create_component_output_skeleton(component, num_samples):
     if hasattr(component, 'lb'):
         comp_dict["lower bound"] = component.lb
     if hasattr(component, 'ub'):
-        comp_dict["upper bound"] = component.lb
+        comp_dict["upper bound"] = component.ub
     if hasattr(component, 'get_units'):
         unit_obj = component.get_units()
         if unit_obj is not None:
             comp_dict["units"] = component.get_units().name
         else:
             comp_dict["units"] = "None"
 
+    # Add information to this output that WILL NOT be written as part
+    # of the file saving step.
+    comp_dict['_pyo_obj'] = component
```

Comment:
I'm a bit worried that we'll be sending Pyomo objects over MPI now. That should generally be avoided -- they have to be pickled and reconstructed on the other end. This could become a major bottleneck on rank 0 for large scale sweeps.

Reply:
We don't send promo objects using MPI. The only items that are collected on rank 0 are numpy arrays.
https://github.com/k1nshuk/watertap/blob/d3ec6653b9fa1ab0b4bab6005cd44d58ae4380b4/watertap/tools/parameter_sweep.py#L446-L448

Those pyomo objects remain in the local output dictionary and are popped off before any h5 files are written.